### PR TITLE
asm/f64: Updated scal assembly to wide, pipelined loops.

### PR DESCRIPTION
**Pull request by Kunde21**
This depends on updates in gonum/internal#52

Cleaned up and widened loops in scal* assembly functions.  

<details><summary>Benchmarks vs old asm</summary>

<table class='benchstat oldnew'>
<tr class='configs'><th><th>Old asm<th>New asm

<tbody>
<tr><th><th colspan='2' class='metric'>time/op<th>delta
<tr class='unchanged'><td>ScalUnitary/ScalUnitary-1-12<td>2.89ns ± 7%<td>2.92ns ±13%<td class='nodelta'>~<td class='note'>(p=0.862 n=19&#43;20)
<tr class='better'><td>ScalUnitary/ScalUnitary-2-12<td>3.03ns ± 1%<td>2.77ns ± 3%<td class='delta'>−8.49%<td class='note'>(p=0.000 n=17&#43;17)
<tr class='better'><td>ScalUnitary/ScalUnitary-3-12<td>3.88ns ±12%<td>3.07ns ± 1%<td class='delta'>−20.74%<td class='note'>(p=0.000 n=20&#43;16)
<tr class='worse'><td>ScalUnitary/ScalUnitary-4-12<td>2.87ns ±13%<td>3.41ns ±10%<td class='delta'>&#43;18.68%<td class='note'>(p=0.000 n=19&#43;19)
<tr class='worse'><td>ScalUnitary/ScalUnitary-5-12<td>3.48ns ±19%<td>3.52ns ± 6%<td class='delta'>&#43;1.29%<td class='note'>(p=0.028 n=20&#43;18)
<tr class='better'><td>ScalUnitary/ScalUnitary-10-12<td>4.75ns ± 3%<td>4.39ns ± 5%<td class='delta'>−7.56%<td class='note'>(p=0.000 n=16&#43;18)
<tr class='better'><td>ScalUnitary/ScalUnitary-100-12<td>20.9ns ± 2%<td>17.2ns ± 3%<td class='delta'>−17.57%<td class='note'>(p=0.000 n=16&#43;17)
<tr class='better'><td>ScalUnitary/ScalUnitary-500-12<td>79.4ns ± 0%<td>71.8ns ± 9%<td class='delta'>−9.53%<td class='note'>(p=0.000 n=16&#43;16)
<tr class='better'><td>ScalUnitary/ScalUnitary-1000-12<td>145ns ± 2%<td>142ns ±14%<td class='delta'>−2.08%<td class='note'>(p=0.031 n=18&#43;20)
<tr class='better'><td>ScalUnitary/ScalUnitary-5000-12<td>1.04µs ± 0%<td>1.03µs ± 1%<td class='delta'>−0.65%<td class='note'>(p=0.000 n=16&#43;16)
<tr class='better'><td>ScalUnitary/ScalUnitary-10000-12<td>2.07µs ± 0%<td>2.05µs ± 0%<td class='delta'>−0.82%<td class='note'>(p=0.000 n=16&#43;17)
<tr class='unchanged'><td>ScalUnitary/ScalUnitary-50000-12<td>15.4µs ± 3%<td>15.6µs ± 4%<td class='nodelta'>~<td class='note'>(p=0.178 n=16&#43;18)
<tr class='unchanged'><td>ScalUnitaryTo/ScalUnitaryTo-1-12<td>3.02ns ± 4%<td>3.09ns ±14%<td class='nodelta'>~<td class='note'>(p=0.840 n=9&#43;9)
<tr class='better'><td>ScalUnitaryTo/ScalUnitaryTo-2-12<td>3.57ns ± 7%<td>3.12ns ± 2%<td class='delta'>−12.65%<td class='note'>(p=0.000 n=10&#43;9)
<tr class='better'><td>ScalUnitaryTo/ScalUnitaryTo-3-12<td>4.07ns ± 1%<td>3.57ns ± 1%<td class='delta'>−12.23%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='worse'><td>ScalUnitaryTo/ScalUnitaryTo-4-12<td>3.33ns ± 0%<td>4.35ns ± 2%<td class='delta'>&#43;30.76%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='worse'><td>ScalUnitaryTo/ScalUnitaryTo-5-12<td>4.05ns ± 0%<td>4.26ns ± 9%<td class='delta'>&#43;5.14%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='unchanged'><td>ScalUnitaryTo/ScalUnitaryTo-10-12<td>5.38ns ± 8%<td>5.23ns ±26%<td class='nodelta'>~<td class='note'>(p=0.102 n=10&#43;10)
<tr class='worse'><td>ScalUnitaryTo/ScalUnitaryTo-100-12<td>17.9ns ± 0%<td>18.1ns ± 2%<td class='delta'>&#43;1.19%<td class='note'>(p=0.001 n=8&#43;8)
<tr class='better'><td>ScalUnitaryTo/ScalUnitaryTo-500-12<td>76.8ns ±10%<td>71.8ns ± 2%<td class='delta'>−6.54%<td class='note'>(p=0.000 n=9&#43;9)
<tr class='better'><td>ScalUnitaryTo/ScalUnitaryTo-1000-12<td>140ns ± 2%<td>134ns ± 1%<td class='delta'>−4.63%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='better'><td>ScalUnitaryTo/ScalUnitaryTo-5000-12<td>1.35µs ±15%<td>1.27µs ± 1%<td class='delta'>−6.07%<td class='note'>(p=0.011 n=10&#43;9)
<tr class='unchanged'><td>ScalUnitaryTo/ScalUnitaryTo-10000-12<td>2.80µs ±20%<td>2.63µs ± 7%<td class='nodelta'>~<td class='note'>(p=0.424 n=10&#43;10)
<tr class='unchanged'><td>ScalUnitaryTo/ScalUnitaryTo-50000-12<td>23.0µs ± 0%<td>22.9µs ± 0%<td class='nodelta'>~<td class='note'>(p=0.200 n=9&#43;8)
<tr class='worse'><td>ScalInc/ScalInc-1-inc(1)-12<td>2.84ns ± 1%<td>2.98ns ± 1%<td class='delta'>&#43;4.66%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='better'><td>ScalInc/ScalInc-2-inc(1)-12<td>3.52ns ± 1%<td>3.41ns ± 1%<td class='delta'>−2.95%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='better'><td>ScalInc/ScalInc-2-inc(2)-12<td>3.56ns ± 7%<td>3.42ns ± 4%<td class='delta'>−4.02%<td class='note'>(p=0.001 n=9&#43;9)
<tr class='unchanged'><td>ScalInc/ScalInc-2-inc(4)-12<td>3.55ns ± 2%<td>3.54ns ±10%<td class='nodelta'>~<td class='note'>(p=0.389 n=9&#43;10)
<tr class='better'><td>ScalInc/ScalInc-2-inc(10)-12<td>3.50ns ± 1%<td>3.41ns ± 1%<td class='delta'>−2.79%<td class='note'>(p=0.000 n=9&#43;9)
<tr class='unchanged'><td>ScalInc/ScalInc-3-inc(1)-12<td>4.03ns ± 8%<td>3.95ns ±11%<td class='nodelta'>~<td class='note'>(p=0.238 n=10&#43;10)
<tr class='better'><td>ScalInc/ScalInc-3-inc(2)-12<td>3.81ns ± 0%<td>3.75ns ± 0%<td class='delta'>−1.66%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='better'><td>ScalInc/ScalInc-3-inc(4)-12<td>3.95ns ±10%<td>3.75ns ± 1%<td class='delta'>−5.09%<td class='note'>(p=0.000 n=10&#43;8)
<tr class='unchanged'><td>ScalInc/ScalInc-3-inc(10)-12<td>3.82ns ± 0%<td>3.81ns ± 7%<td class='nodelta'>~<td class='note'>(p=0.053 n=8&#43;9)
<tr class='better'><td>ScalInc/ScalInc-4-inc(1)-12<td>4.54ns ± 4%<td>4.23ns ± 5%<td class='delta'>−6.83%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='better'><td>ScalInc/ScalInc-4-inc(2)-12<td>4.54ns ± 2%<td>4.21ns ± 2%<td class='delta'>−7.32%<td class='note'>(p=0.000 n=9&#43;9)
<tr class='better'><td>ScalInc/ScalInc-4-inc(4)-12<td>4.56ns ± 4%<td>4.25ns ± 5%<td class='delta'>−6.85%<td class='note'>(p=0.000 n=9&#43;9)
<tr class='better'><td>ScalInc/ScalInc-4-inc(10)-12<td>4.50ns ± 0%<td>4.19ns ± 1%<td class='delta'>−6.97%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='better'><td>ScalInc/ScalInc-5-inc(1)-12<td>5.13ns ± 1%<td>4.96ns ± 0%<td class='delta'>−3.24%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='better'><td>ScalInc/ScalInc-5-inc(2)-12<td>5.37ns ±11%<td>4.98ns ± 1%<td class='delta'>−7.35%<td class='note'>(p=0.000 n=10&#43;8)
<tr class='better'><td>ScalInc/ScalInc-5-inc(4)-12<td>5.22ns ± 3%<td>5.07ns ± 5%<td class='delta'>−2.85%<td class='note'>(p=0.038 n=9&#43;9)
<tr class='unchanged'><td>ScalInc/ScalInc-5-inc(10)-12<td>5.11ns ± 0%<td>5.20ns ± 9%<td class='nodelta'>~<td class='note'>(p=0.712 n=9&#43;9)
<tr class='worse'><td>ScalInc/ScalInc-10-inc(1)-12<td>8.01ns ± 4%<td>8.85ns ±14%<td class='delta'>&#43;10.49%<td class='note'>(p=0.005 n=8&#43;10)
<tr class='unchanged'><td>ScalInc/ScalInc-10-inc(2)-12<td>8.35ns ±22%<td>8.10ns ± 1%<td class='nodelta'>~<td class='note'>(p=0.243 n=9&#43;9)
<tr class='unchanged'><td>ScalInc/ScalInc-10-inc(4)-12<td>9.10ns ±14%<td>8.13ns ± 4%<td class='nodelta'>~<td class='note'>(p=0.479 n=10&#43;9)
<tr class='unchanged'><td>ScalInc/ScalInc-10-inc(10)-12<td>8.70ns ±16%<td>8.12ns ± 0%<td class='nodelta'>~<td class='note'>(p=0.501 n=10&#43;8)
<tr class='better'><td>ScalInc/ScalInc-500-inc(1)-12<td>457ns ±17%<td>133ns ± 2%<td class='delta'>−70.97%<td class='note'>(p=0.000 n=10&#43;8)
<tr class='better'><td>ScalInc/ScalInc-500-inc(2)-12<td>435ns ± 3%<td>136ns ± 9%<td class='delta'>−68.79%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='better'><td>ScalInc/ScalInc-500-inc(4)-12<td>444ns ±13%<td>139ns ±15%<td class='delta'>−68.75%<td class='note'>(p=0.000 n=9&#43;10)
<tr class='better'><td>ScalInc/ScalInc-500-inc(10)-12<td>447ns ± 8%<td>152ns ± 1%<td class='delta'>−66.12%<td class='note'>(p=0.000 n=9&#43;8)
<tr class='better'><td>ScalInc/ScalInc-1000-inc(1)-12<td>924ns ±17%<td>259ns ± 1%<td class='delta'>−72.01%<td class='note'>(p=0.000 n=10&#43;9)
<tr class='better'><td>ScalInc/ScalInc-1000-inc(2)-12<td>940ns ±12%<td>267ns ± 8%<td class='delta'>−71.55%<td class='note'>(p=0.000 n=10&#43;10)
<tr class='better'><td>ScalInc/ScalInc-1000-inc(4)-12<td>946ns ±22%<td>285ns ± 9%<td class='delta'>−69.87%<td class='note'>(p=0.000 n=10&#43;9)
<tr class='worse'><td>ScalInc/ScalInc-1000-inc(10)-12<td>1.61µs ± 0%<td>1.63µs ± 0%<td class='delta'>&#43;1.15%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='better'><td>ScalInc/ScalInc-10000-inc(1)-12<td>8.76µs ± 0%<td>3.07µs ± 0%<td class='delta'>−64.98%<td class='note'>(p=0.000 n=9&#43;10)
<tr class='better'><td>ScalInc/ScalInc-10000-inc(2)-12<td>9.32µs ± 9%<td>4.46µs ±19%<td class='delta'>−52.11%<td class='note'>(p=0.000 n=10&#43;10)
<tr class='unchanged'><td>ScalInc/ScalInc-10000-inc(4)-12<td>11.1µs ± 4%<td>11.1µs ±12%<td class='nodelta'>~<td class='note'>(p=0.604 n=9&#43;10)
<tr class='unchanged'><td>ScalInc/ScalInc-10000-inc(10)-12<td>24.3µs ± 9%<td>24.8µs ±17%<td class='nodelta'>~<td class='note'>(p=0.725 n=10&#43;10)
<tr class='worse'><td>ScalIncTo/ScalIncTo-1-inc(1)-12<td>3.77ns ± 3%<td>3.93ns ±10%<td class='delta'>&#43;4.04%<td class='note'>(p=0.003 n=9&#43;10)
<tr class='better'><td>ScalIncTo/ScalIncTo-2-inc(1)-12<td>4.56ns ± 4%<td>4.49ns ± 5%<td class='delta'>−1.65%<td class='note'>(p=0.023 n=9&#43;10)
<tr class='better'><td>ScalIncTo/ScalIncTo-2-inc(2)-12<td>4.52ns ± 1%<td>4.43ns ± 0%<td class='delta'>−2.09%<td class='note'>(p=0.000 n=9&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-2-inc(4)-12<td>4.53ns ± 1%<td>4.42ns ± 0%<td class='delta'>−2.35%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-2-inc(10)-12<td>4.84ns ±23%<td>4.43ns ± 1%<td class='delta'>−8.43%<td class='note'>(p=0.000 n=10&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-3-inc(1)-12<td>4.99ns ± 4%<td>4.76ns ± 0%<td class='delta'>−4.68%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-3-inc(2)-12<td>4.97ns ± 1%<td>4.77ns ± 1%<td class='delta'>−4.19%<td class='note'>(p=0.000 n=9&#43;8)
<tr class='unchanged'><td>ScalIncTo/ScalIncTo-3-inc(4)-12<td>4.97ns ± 1%<td>4.89ns ± 9%<td class='nodelta'>~<td class='note'>(p=0.163 n=8&#43;10)
<tr class='better'><td>ScalIncTo/ScalIncTo-3-inc(10)-12<td>5.08ns ±12%<td>4.76ns ± 0%<td class='delta'>−6.30%<td class='note'>(p=0.000 n=9&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-4-inc(1)-12<td>5.76ns ± 4%<td>5.19ns ± 5%<td class='delta'>−9.87%<td class='note'>(p=0.000 n=9&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-4-inc(2)-12<td>5.76ns ± 3%<td>5.13ns ± 0%<td class='delta'>−11.01%<td class='note'>(p=0.000 n=9&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-4-inc(4)-12<td>6.02ns ±14%<td>5.13ns ± 1%<td class='delta'>−14.73%<td class='note'>(p=0.000 n=10&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-4-inc(10)-12<td>5.72ns ± 1%<td>5.19ns ± 3%<td class='delta'>−9.15%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-5-inc(1)-12<td>6.08ns ± 1%<td>5.60ns ± 0%<td class='delta'>−8.01%<td class='note'>(p=0.000 n=8&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-5-inc(2)-12<td>6.12ns ± 3%<td>5.72ns ± 3%<td class='delta'>−6.48%<td class='note'>(p=0.000 n=8&#43;10)
<tr class='better'><td>ScalIncTo/ScalIncTo-5-inc(4)-12<td>6.07ns ± 0%<td>5.64ns ± 2%<td class='delta'>−7.17%<td class='note'>(p=0.000 n=9&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-5-inc(10)-12<td>6.07ns ± 0%<td>5.67ns ± 2%<td class='delta'>−6.67%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='unchanged'><td>ScalIncTo/ScalIncTo-10-inc(1)-12<td>9.54ns ±16%<td>9.26ns ± 0%<td class='nodelta'>~<td class='note'>(p=0.496 n=10&#43;8)
<tr class='worse'><td>ScalIncTo/ScalIncTo-10-inc(2)-12<td>9.02ns ± 1%<td>10.06ns ±22%<td class='delta'>&#43;11.63%<td class='note'>(p=0.000 n=8&#43;10)
<tr class='worse'><td>ScalIncTo/ScalIncTo-10-inc(4)-12<td>9.06ns ± 2%<td>9.28ns ± 1%<td class='delta'>&#43;2.42%<td class='note'>(p=0.001 n=9&#43;8)
<tr class='unchanged'><td>ScalIncTo/ScalIncTo-10-inc(10)-12<td>9.63ns ±11%<td>9.27ns ± 1%<td class='nodelta'>~<td class='note'>(p=0.498 n=10&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-500-inc(1)-12<td>457ns ±13%<td>136ns ± 1%<td class='delta'>−70.34%<td class='note'>(p=0.000 n=10&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-500-inc(2)-12<td>436ns ± 0%<td>135ns ± 0%<td class='delta'>−69.06%<td class='note'>(p=0.000 n=9&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-500-inc(4)-12<td>440ns ± 0%<td>219ns ± 4%<td class='delta'>−50.23%<td class='note'>(p=0.000 n=8&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-500-inc(10)-12<td>1.00µs ± 0%<td>1.00µs ± 0%<td class='delta'>−0.24%<td class='note'>(p=0.012 n=9&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-1000-inc(1)-12<td>875ns ± 0%<td>261ns ± 0%<td class='delta'>−70.16%<td class='note'>(p=0.000 n=9&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-1000-inc(2)-12<td>878ns ± 0%<td>362ns ± 8%<td class='delta'>−58.72%<td class='note'>(p=0.000 n=8&#43;10)
<tr class='better'><td>ScalIncTo/ScalIncTo-1000-inc(4)-12<td>1.01µs ± 1%<td>1.01µs ± 1%<td class='delta'>−0.83%<td class='note'>(p=0.000 n=9&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-1000-inc(10)-12<td>2.09µs ± 9%<td>2.03µs ± 4%<td class='delta'>−2.89%<td class='note'>(p=0.013 n=10&#43;9)
<tr class='better'><td>ScalIncTo/ScalIncTo-10000-inc(1)-12<td>8.77µs ± 0%<td>3.35µs ± 1%<td class='delta'>−61.81%<td class='note'>(p=0.000 n=9&#43;8)
<tr class='better'><td>ScalIncTo/ScalIncTo-10000-inc(2)-12<td>9.16µs ± 1%<td>7.16µs ± 6%<td class='delta'>−21.90%<td class='note'>(p=0.000 n=9&#43;10)
<tr class='unchanged'><td>ScalIncTo/ScalIncTo-10000-inc(4)-12<td>19.0µs ±11%<td>18.7µs ± 8%<td class='nodelta'>~<td class='note'>(p=0.529 n=10&#43;10)
<tr class='unchanged'><td>ScalIncTo/ScalIncTo-10000-inc(10)-12<td>38.5µs ± 0%<td>38.5µs ± 1%<td class='nodelta'>~<td class='note'>(p=0.624 n=8&#43;8)
<tr><td>&nbsp;
</tbody>

</table>

</details>

### CODE REVIEW

File: asm/f64/asm_test.go [hunk 1]
```diff
@@ -65,3 +74,119 @@ func nonStridedWrite(x []float64, inc int) bool {
...
+		guarded[i] = gdVal
+		guarded[len(guarded)-1-i] = gdVal
+	}
+	return guarded
+}
+
+// isValidGuard will test for violated guards, generated by guardVector.
+func isValidGuard(vec []float64, gdVal float64, gdLn int) bool {
+	for i := 0; i < gdLn; i++ {
+		if !same(vec[i], gdVal) || vec[len(vec)-1-i] != gdVal {
```
Comment (vladimir-ch) [inline asm/f64/asm_test.go]: Use `same` also at the end of `vec`?

**Review by vladimir-ch — COMMENTED** [review]
Went through the first 3 commits, will continue later.

File: asm/f64/asm_test.go (hunk 1 above)
Comment (Kunde21) [reply asm/f64/asm_test.go]: Done.  Fix at [commit on 54](https://github.com/gonum/internal/pull/54/commits/60458ee82ea864ec4eebc5e368eda4fce6d92821)

File: asm/f64/axpyunitaryto_amd64.s [hunk 2]
```diff
@@ -38,51 +38,103 @@
...
+	ADDPD 16(Y_PTR)(IDX*8), X3
+	ADDPD 32(Y_PTR)(IDX*8), X4
+	ADDPD 48(Y_PTR)(IDX*8), X5
+
+	MOVUPS X2, (DST_PTR)(IDX*8)   // y[i] = X_i
+	MOVUPS X3, 16(DST_PTR)(IDX*8)
+	MOVUPS X4, 32(DST_PTR)(IDX*8)
+	MOVUPS X5, 48(DST_PTR)(IDX*8)
+
+	ADDQ $8, IDX  // i += 2
```
Comment (vladimir-ch) [inline asm/f64/axpyunitaryto_amd64.s]: `+= 8`

File: asm/f64/axpyunitaryto_amd64.s [hunk 3]
```diff
@@ -38,51 +38,103 @@
...
+	JE   end     // if LEN == 0 { return }
+
+	XORQ   IDX, IDX            // IDX = 0
+	MOVSD  alpha+24(FP), ALPHA
+	SHUFPD $0, ALPHA, ALPHA    // ALPHA := { alpha, alpha }
+	MOVQ   Y_PTR, TAIL         // Check memory alignment
+	ANDQ   $15, TAIL           // TAIL = &y % 16
+	JZ     no_trim             // if TAIL == 0 { goto no_trim }
+
+	// Align on 16-bit boundary
```
Comment (vladimir-ch) [inline asm/f64/axpyunitaryto_amd64.s]: 16-byte

File: asm/f64/axpyunitary_amd64.s [hunk 4]
```diff
@@ -38,50 +38,97 @@
...
+	JE      end
+	XORQ    IDX, IDX
+	MOVSD   alpha+0(FP), ALPHA   // ALPHA := { alpha, alpha }
+	SHUFPD  $0, ALPHA, ALPHA
+	MOVUPS  ALPHA, ALPHA_2       // ALPHA_2 := ALPHA   for pipelining
+	MOVQ    Y_PTR, TAIL          // Check memory alignment
+	ANDQ    $15, TAIL            // TAIL = &y % 16
+	JZ      no_trim              // if TAIL == 0 { goto no_trim }
+
+	// Align on 16-bit boundary
```
Comment (vladimir-ch) [inline asm/f64/axpyunitary_amd64.s]: 16-byte

**Review by vladimir-ch — COMMENTED** [review]
Another commit reviewed.

**Review by vladimir-ch — APPROVED** [review]
Sorry it took me so long, I need smaller PRs. Please wait also for @kortschak .

**Comment by Kunde21** [comment]
All of @vladimir-ch's comments corrected in [commit on 54](https://github.com/gonum/internal/pull/54/commits/68a7a5322803c84dc6f49fe40a0fe253430c401e).

File: asm/f64/asm_test.go (hunk 1 above)
Comment (kortschak) [reply asm/f64/asm_test.go]: This is not done.

Comment (Kunde21) [reply asm/f64/asm_test.go]: I linked the commit in which this was fixed.  It's from a commit on gonum/internal#54, so it's fixed on that PR.

Comment (kortschak) [reply asm/f64/asm_test.go]: Ah, OK. Sorry.

File: asm/f64/benchScal_test.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
```
Comment (kortschak) [inline asm/f64/benchScal_test.go]: 2017?

**Review by kortschak — APPROVED** [review]

**Pull request merged**